src/test/java/de/bonndan/nivio/input/GroupQueryResolverTest.java

```diff
@@ -0,0 +1,97 @@
+package de.bonndan.nivio.input;
+
+import de.bonndan.nivio.input.dto.GroupDescription;
+import de.bonndan.nivio.input.dto.LandscapeDescription;
+import de.bonndan.nivio.model.Group;
+import de.bonndan.nivio.model.Item;
+import de.bonndan.nivio.model.LandscapeImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.List;
+import java.util.Set;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@ExtendWith(MockitoExtension.class)
+class GroupQueryResolverTest {
+
+    private GroupQueryResolver groupResolver;
+
+    @Mock
+    private ProcessLog processLog;
+
+    @BeforeEach
+    public void setup() {
+        groupResolver = new GroupQueryResolver(processLog);
+    }
+
+    @Test
+    void process_doesNotContainDuplicatedItemsInGroups() {
+        LandscapeImpl landscape = new LandscapeImpl();
+        landscape.setProcessLog(processLog);
+        Item item = new Item();
+        item.setIdentifier("itemIdentifier");
+        item.setGroup("groupIdentifier");
+        landscape.setItems(Set.of(item));
+        landscape.addGroup(new Group("groupIdentifier"));
+
+        LandscapeDescription input = new LandscapeDescription();
+        GroupDescription groupDescription = new GroupDescription();
+        groupDescription.setIdentifier("groupIdentifier");
+        groupDescription.setContains(List.of("itemidentifier"));
+        input.getGroups().put("group", groupDescription);
+
+        groupResolver.process(input, landscape);
```

Comment:
This shows what has to be done to have valid objects (instantiation and setting of things), only to not have a NPE. 

I am not even sure if this is how the objects are intended to be used 😞, which means that somehow it is not straightforward and thus potentially unsafe.

I also did not really understand why we have `Group` and `GroupDescription` both implementing `GroupItem`. There are several casts to these objects, which are really hard to get rid of.

Reply:
Please see LandscapeFactory, it now sets up a proper landscape. We might think about more safety measures though.

Concerning Group and GroupDescription: #212 .